### PR TITLE
Add a forwardRef to the Input component

### PR DESCRIPTION
**Pull request by jasperswart**
In this PR:
 - added ref to Input component
- added test

### CODE REVIEW

File: packages/asc-ui/src/components/Input/Input.tsx [hunk 1]
```diff
@@ -23,62 +23,63 @@ export interface InputProps extends InputMethods {
   error?: string
 }
 
-export interface Shared {}
+const Input = React.forwardRef<
+  HTMLInputElement,
+  InputProps & React.HTMLAttributes<HTMLInputElement>
+>(
+  (
+    { blurOnEscape, focusOnRender, value, error, ...props }: InputProps,
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/Input/Input.tsx]: If i'm not mistaken, you can remove the  :InputProps and the :any, as these should already be typed

Comment (jasperswart) [reply packages/asc-ui/src/components/Input/Input.tsx]: dankjewel

Comment (jasperswart) [reply packages/asc-ui/src/components/Input/Input.tsx]: gedaan

File: packages/asc-ui/src/components/Input/Input.tsx [hunk 2]
```diff
@@ -28,7 +28,16 @@ const Input = React.forwardRef<
...
   const internalRef = React.useRef<HTMLInputElement>(null)
+  React.useImperativeHandle<HTMLInputElement, any>(externalRef, () => ({
+    blur: () => {
+      if (internalRef.current) {
+        internalRef.current.blur()
+      }
+    },
+  }))
+
   const ref = externalRef || internalRef
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/Input/Input.tsx]: So this line can be removed, and `internalRef` can be renamed to `ref`

**Comment by jonkoops** [comment]
Looks like we we're using `useImperativeHandle` in the wrong manner, I've changed the implementation and squashed the commits. Tests seem to pass properly now as they are indeed recieving a `HTMLInputElement` as expected.

**Pull request merged**